### PR TITLE
Return status code on successful delete

### LINKED ISSUE #86: TimeSync DELETE returns empty body, which confuses pymesync
When delete something, pymesync gives you this:

```
[{'pymesync error': 'connection to TimeSync failed at baseurl http://timesync-staging.osuosl.org/v1 - response status was 200'}]
```

because the response isn't json!

We need to check of an empty response.text with a status code of 200, then just return something like

```
[{'status': 200}]
```

### PR DESCRIPTION
**Pull request by matthewrsj**
Fixes #86

### CODE REVIEW

File: pymesync.py [hunk 1]
```diff
@@ -535,6 +535,10 @@ def __local_auth_error(self):
 
     def __response_to_python(self, response):
         """Convert response to native python list of objects"""
+        # DELETE returns an empty body if successful
+        if not response.text and response.status_code == 200:
```
Comment (ElijahCaine) [inline pymesync.py]: Why do you want response.text to be None?

Comment (matthewrsj) [inline pymesync.py]: @ElijahCaine response.text is None upon successful DELETE requests according to https://github.com/osuosl/timesync/blob/develop/source/draft_api.rst#delete-endpoints . I also checked that this code check works with the staging instance we have up for timesync-node.

Comment (ElijahCaine) [inline pymesync.py]: @matthewrsj Hmm, this seems like a bad function name if it's only used for DELETES. If you could change the name or change the function to reflect the name/docstring it'd sit better with me.

``__response_to_python`` feels like a method which always converts a response to a python data-type.

Comment (matthewrsj) [inline pymesync.py]: @ElijahCaine It isn't only used for deletes, this if statement is what handles what TimeSync returns upon a successful DELETE. The rest of the method covers all responses. It takes a response and returns a python dict. The response doesn't only include the text field, but also status_code and some others I don't use.

**Comment by Kennric** [comment]
@ElijahCaine - yeah, that's basically what the method does - if you look at the whole module, you see it's handing responses from top level methods.

I think in future we may revisit the DELETE response bodies in the API (now that things are soft-deleted) but for now I think this is good. +1

**Pull request merged**